domain/objectstore/state/state.go

```diff
@@ -138,13 +137,15 @@ FROM   object_store_metadata
 WHERE  hash = $dbMetadata.hash 
 AND    size = $dbMetadata.size`, dbMetadata, dbMetadataPath)
 	if err != nil {
-		return errors.Annotate(err, "preparing select metadata statement")
+		return "", errors.Annotate(err, "preparing select metadata statement")
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		var outcome sqlair.Outcome
 		err := tx.Query(ctx, metadataStmt, dbMetadata).Get(&outcome)
-		if err != nil {
+		if database.IsErrConstraintPrimaryKey(err) {
+			return objectstoreerrors.ErrHashAlreadyExists
```

Comment:
Is there a test for this error now? Looks like it wouldn't have been returned before, but that should probably have been caught by a test.

Reply:
Turns out we're leaking errors, even though they're coerced.

Reply:
Haha, exciting. I wonder how much code is clutching on this